### PR TITLE
[Tizen/API] Fix pipeline-src implementation @open sesame 4/10 10:22

### PR DESCRIPTION
**Pull request by myungjoo**
- Return TensorsInfo with get-src-handle
- Fix src-inputdata with TensorsInfo caching mechanism
    
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

The second commit in this PR is to add a unit-test-case to prevent regression.

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tizen-api/src/tizen-api-pipeline.c [hunk 1]
```diff
@@ -630,6 +630,45 @@ int nns_pipeline_src_gethandle
...
+  if (elem->src == NULL)
+    elem->src = gst_element_get_static_pad (elem->element, "src");
+
+  if (elem->src != NULL) {
+    /** @todo : refactor this along with nns_pipeline_src_inputdata */
+    GstCaps *caps = gst_pad_get_allowed_caps (elem->src);
+
+    /** @todo caps may be NULL for prerolling */
+    if (caps == NULL) {
+      g_printerr ("Cannot find caps...\n");
```
Comment (again4you) [inline tizen-api/src/tizen-api-pipeline.c]: Is `g_printerr()` right instead of `dlog_print()`?

File: tests/tizen_capi/unittest_tizen_capi.cpp [hunk 2]
```diff
@@ -231,6 +233,240 @@ TEST (nnstreamer_capi_valve, test01)
...
 }
 
+/**
+ * @brief A tensor-sink callback for sink handle in a pipeline
+ */
+static void nns_sink_callback_dm01 (const char *buf[], const size_t size[],
+    const GstTensorsInfo *tensorsinfo, void *pdata)
+{
+  gchar *filepath = (gchar *) pdata;
+  FILE *fp = g_fopen (filepath, "a");
```
Comment (kparichay) [inline tests/tizen_capi/unittest_tizen_capi.cpp]:
```suggestion
  FILE *fp = g_fopen (filepath, "a");
  if (fp == NULL) {
    return;
  }
```

File: tests/tizen_capi/unittest_tizen_capi.cpp [hunk 3]
```diff
@@ -231,6 +233,240 @@ TEST (nnstreamer_capi_valve, test01)
...
+  gchar *file2 = g_build_path ("/", dir, "sink", NULL);
+  gchar *pipeline = g_strdup_printf ("videotestsrc num-buffers=3 ! videoconvert ! video/x-raw,format=BGRx,width=64,height=48,famerate=60/1 ! tee name=t t. ! queue ! filesink location=\"%s\"  t. ! queue ! tensor_converter ! tensor_sink name=sinkx",
+      file1);
+  nns_pipeline_h handle;
+  nns_pipeline_state_e state;
+  nns_sink_h sinkhandle;
+  int status = nns_pipeline_construct (pipeline, &handle);
+  EXPECT_EQ (status, NNS_ERROR_NONE);
+
+  EXPECT_TRUE (dir != NULL);
```
Comment (kparichay) [inline tests/tizen_capi/unittest_tizen_capi.cpp]: Insert this check after line 293
Also, should be ASSERT() as EXPECT will continue on failure.

Comment (myungjoo) [reply tests/tizen_capi/unittest_tizen_capi.cpp]: Updated.

File: tests/tizen_capi/unittest_tizen_capi.cpp (hunk 2 above)
Comment (myungjoo) [reply tests/tizen_capi/unittest_tizen_capi.cpp]: Updated.

**Pull request merged**